### DISCUSSION TITLE
Object masks - how do they work with objects in motion?

### DISCUSSION THREAD

**Discussion by LaurenceGough** [blakeblackshear/frigate#5917] [Ask A Question] [answered]
Hello all,

Hopefully someone has experience with this.

I am having to use a lot of object masks, many to avoid a lot of false positives I'm having. 

How does it work if an object (person for example) starts being detected and then moves over this object (person) mask? Would it flat out avoid registering that as an event?  

In reality the person would be tracked across most of the frame in the majority of cases, I was just wondering if they were to walk into one of these object masks during the tracking if the whole event / tracking would be cancelled.

I will give it a go and report back if no one has experience of testing this :).

Thank you

**Comment by NickM-27** [comment, marked as answer]
The mask simply makes that detection in that single frame be ignored. So it would be like the person disappeared for that frame, if they stay in the mask more frames than your `detect -> max_disappeared` then the event would end. If the object was in the mask then exited it would continue the event as is.

**Reply by LaurenceGough** [reply]
Thanks again Nick :). Ok hopefully events won't be "duplicated" as they pass the object mask area, that is my only fear now but it is what it is, and even if that does happen it's still far less of an issue than nothing being detected of course :).

I believe sometimes events can be duplicated if the tracking of an object is interrupted. I'm sure I was reading a thread on this about a car object going past pillars as an example. A very tricky thing of course for object recognition to deal with, although in this case, technically it'll be able to see the object at all times, and it is just a virtual object mask.
